DESIGN.md

```diff
@@ -0,0 +1,279 @@
+# Fleetspeak Design
+
+Fleetspeak is a client-server framework for communicating with a fleet of
+machines. This document summarizes the various components, providing an overview
+of what bits serve what purpose.
+
+## Identity
+
+Fleetspeak identifies clients using a fingerprint of a public key, much like
+openssl keys. Specifically, client creates a keypair on startup, or when asked
+to rekey. The Fleetspeak
+[common.ClientID](https://godoc.org/github.com/google/fleetspeak/fleetspeak/src/common#ClientID)
+the first 8 bytes of a sha256 hash of the public part of this key. It is the
```

Comment:
Doesn't grammar. "common.clientID holds the first 8 bytes"?

Reply:
"is" was the missing verb, already fixed.